src/arch/x86/image/multiboot2.c

```
@@ -461,6 +463,173 @@ static int multiboot2_add_bootloader ( struct mb2 *mb2 ) {
 	return 0;
 }
 
+static EFI_STATUS efifb_colour_map_mask ( uint32_t mask, uint8_t *scale,
+					uint8_t *lsb ) {
+	uint32_t check;
+
+	/* Fill in LSB and scale */
+	*lsb = ( mask ? ( ffs ( mask ) - 1 ) : 0 );
+	*scale = fls ( mask >> *lsb );
+
+	/* Check that original mask was contiguous */
+	check = ( ( 0xff >> ( 8 - *scale ) ) << *lsb );
+	if ( check != mask )
+		return EFI_UNSUPPORTED;
+
+	return EFI_SUCCESS;
+}
+
+#if !defined(EFI_CONSOLE_OUT_DEVICE_GUID)
+#define	EFI_CONSOLE_OUT_DEVICE_GUID	\
```

Could you cite a source on this?  E.g.

https://uefi.org/specs/UEFI/2.9_A/12_Protocols_Console_Support.html

---

It is actually complicated -- this specific GUID is not universally available, hence we need to keep alternate option (and not error out when not available). But it was found out usable at least on some hardware.